builder/docker.go

```diff
@@ -23,14 +23,33 @@ func loadDockerManifest(ctx context.Context, console conslogging.ConsoleLogger,
 	if len(children) == 0 {
 		return errors.Errorf("no images in manifest list for %s", parentImageName)
 	}
-	// Check if any child has the platform as the default platform (use the first one if none found).
+
+	// Check if any child has the platform as the default platform
 	defaultChild := 0
+	foundPlatform := false
 	for i, child := range children {
 		if child.platform == platutil.DefaultPlatform {
 			defaultChild = i
+			foundPlatform = true
 			break
 		}
 	}
+	if !foundPlatform {
+		// otherwise, look for a platform matching the user's platform
+		for i, child := range children {
+			if child.platform == platutil.UserPlatform {
```

Comment:
We generally don't default to the user's platform, but to the native platform. At least with the `--new-platform` flag.